state/protocol/badger/state.go

```diff
@@ -3,75 +3,214 @@
 package badger
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/dgraph-io/badger/v2"
 
+	"github.com/onflow/flow-go/state/protocol"
+
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module"
-	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/storage"
 	"github.com/onflow/flow-go/storage/badger/operation"
 )
 
 type State struct {
-	metrics  module.ComplianceMetrics
-	tracer   module.Tracer
-	db       *badger.DB
-	headers  storage.Headers
-	seals    storage.Seals
-	index    storage.Index
-	payloads storage.Payloads
-	blocks   storage.Blocks
-	epoch    struct {
+	metrics module.ComplianceMetrics
+	db      *badger.DB
+	headers storage.Headers
+	blocks  storage.Blocks
+	seals   storage.Seals
+	epoch   struct {
 		setups   storage.EpochSetups
 		commits  storage.EpochCommits
 		statuses storage.EpochStatuses
 	}
-	consumer protocol.Consumer
-	cfg      Config
 }
 
-// NewState initializes a new state backed by a badger database, applying the
-// optional configuration parameters.
-func NewState(
+func Bootstrap(
```

Comment:
comment for other reviewers:
* Instead of the bootstrapping operation being part of the `Mutator` https://github.com/onflow/flow-go/blob/115ccee9cae920ee5bc309537cf43a9b152a1cf9/state/protocol/badger/mutator.go#L21 it is now the constructor of the (read-only) `protocol.State`. 
* the method body from the original mutator has been separated into:
   - checking the validity of the provided bootstrapping information ([old code](https://github.com/onflow/flow-go/blob/115ccee9cae920ee5bc309537cf43a9b152a1cf9/state/protocol/badger/mutator.go#L24-L90)) is not performed by the `StateRoot`: https://github.com/onflow/flow-go/blob/32b363388cc2aca861d3a46d23a2e7a1ed801377/state/protocol/badger/state_root.go#L30-L31
   - inserting the bootstrapping information into the data base ([old code](https://github.com/onflow/flow-go/blob/115ccee9cae920ee5bc309537cf43a9b152a1cf9/state/protocol/badger/mutator.go#L92-L178)) has moved here. Except for moving it, there are _not algorithmic changes_ to this part of the code.